.github/workflows/my-cd-pipeline.yml

```diff
@@ -0,0 +1,30 @@
+name: Continuous Deployment
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - 'README.md'
+
+jobs:
+  build:
+    name: Deploy into production
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: login to dockerhub
+      run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
+    - name: build production docker image
+      run: docker build --target production --tag ${{ secrets.DOCKER_LOGIN }}/todo-app-production:${GITHUB_SHA::6} .
+    - name: push build to dockerhub
+      run: docker push ${{ secrets.DOCKER_LOGIN }}/todo-app-production:${GITHUB_SHA::6}
+    - name: pull production image from dockerhub
+      run: docker pull ${{ secrets.DOCKER_LOGIN }}/todo-app-production:${GITHUB_SHA::6}
+    - name: tag production image for deployment to heroku
+      run: docker tag ${{ secrets.DOCKER_LOGIN }}/todo-app-production:${GITHUB_SHA::6} registry.heroku.com/todo-app-devops-academy/web
+    - name: login to docker on the heroku registry
+      run: docker login --username=_ --password=${{ secrets.HEROKU_TOKEN }} registry.heroku.com
+    - name: push production image to heroku registry
+      run: docker push registry.heroku.com/todo-app-devops-academy/web
+    - name: release the container on the heroku registry
+      run: HEROKU_API_KEY=${{ secrets.HEROKU_TOKEN }} heroku container:release -a todo-app-devops-academy web
```

Comment:
Nice solution! An alternative way to lay this out would be to add an `env` section to this build job, something like:
```
    runs-on: ubuntu-latest
    env:
      HEROKU_API_KEY: "${{ secrets.HEROKU_TOKEN }}"
    steps:
      <...>
```
That way the variable would be defined for all future steps